### PR TITLE
Gh847

### PR DESCRIPTION
**Pull request by maxwellchandler**
https://github.com/pepfar-datim/COP-Target-Setting/issues/847

### CODE REVIEW

File: R/dhis2api.R [hunk 1]
```diff
@@ -467,11 +469,10 @@ GetData_DataPack <- function(parameters,
   dimensions <- dplyr::bind_rows(dimensions, dimension_disaggs)
 
   non_mil_types_of_org_units <- 
-    datapackcommons::getMetadata("dimensions", 
-                                 "id:eq:mINJi7rR1a6", 
-                                 "items[name,id]",
-                                 base_url = base_url) %>% 
-    tidyr::unnest(c("items")) %>% 
+    datimutils::getMetadata("dimensions",
```
Comment (sam-bao) [inline R/dhis2api.R:472]: As with the other pull request #37 please use getDimensions instead of getMetadata.

File: R/dhis2api.R [hunk 2]
```diff
@@ -274,9 +276,9 @@ ValidateCodeIdPairs <- function(base_url, codes, ids, type){
                           length(codes) == length(ids))
   original <- tibble::tibble(code = codes, id = ids) %>% unique()
   ids_csv <-  ids %>% unique() %>% paste0(collapse = ",")
-  response <- datapackcommons::getMetadata(type, 
-                                           filters = glue::glue("id:in:[{ids_csv}]"), 
-                                           fields = "id,code", 
+  response <- datimutils::getMetadata(type,
```
Comment (sam-bao) [inline R/dhis2api.R:279]: Does this require `!!`?

File: R/dhis2api.R [hunk 3]
```diff
@@ -150,18 +151,19 @@ GetCountryLevels <- function(base_url, countries_req = NULL){
...
-    plyr::laply(utils::URLencode, reserved = TRUE) %>% 
-    paste0(collapse = ",") %>% 
-    paste0("name:in:[", .,"]") %>%  c("level:eq:4") %>% 
-    getMetadata("organisationUnits", ., base_url = base_url)
+  level_4_countries <- datimutils::getMetadata(organisationUnits,
+                          name %.in% countries$country_name,
+                          level %.in% "4",
+                          fields = "name,id",
+                          as_vector = T,
+                          base_url = "https://datim.org/")
```
Comment (sam-bao) [inline R/dhis2api.R:166]: datim.org shouldn't be hard coded in here.

File: R/dhis2api.R (hunk 2 above)
Comment (maxwellchandler) [reply R/dhis2api.R:279]: @sam-bao Yes, otherwise it will literally pass in "type" into the url

Comment (maxwellchandler) [reply R/dhis2api.R:279]: @sam-bao Ah yes sorry, I got confused and forgot to fix it here because I tried to do it in datimutils before.

**Pull request merged**